chalk-parse/src/ast.rs

```diff
@@ -290,3 +292,47 @@ pub enum Goal {
     // Additional kinds of goals:
     Leaf(LeafGoal),
 }
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+/// This allows users to add arbitrary `A :- B` clauses into the
+/// logic; it has no equivalent in Rust, but it's useful for testing.
+pub struct Scalar {
+    // Some scalars have names if we include variables...
+    // TODO: review this
+    pub name: Option<Identifier>,
+    // Do we want to store the value too?
+    // storing the value is simple if we're already parsing. TODO: is this useful?
+    pub value: ScalarValue,
+}
+
+#[derive(Clone, PartialEq, Eq, Debug)]
+/// The inner value and type of the scalar value.
+pub enum ScalarValue {
```

Comment:
I would leave out scalar values for now -- we won't need these until the const generics stuff lands, and I think it'd be better to add in that PR